drivers/timer/nrf_rtc_timer.c

```diff
@@ -20,8 +20,6 @@
 		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
 #define MAX_TICKS ((COUNTER_MAX - CYC_PER_TICK) / CYC_PER_TICK)
```

Comment:
what if `CYC_PER_TICK = 1`? Then `MAX_TICKS = 0xffffff - 1`. Wouldn't it be simpler to set it to fixed `COUNTER_MAX/2`. Especially that there are some calculations there assuming that when subtraction gives result > 0x7fffff then counter is behind.